### DISCUSSION TITLE
Long running renovate bot executions run into HTTP 403 after 60 min when interacting with Google Artifact Registry

### DISCUSSION THREAD

**Discussion by rvillane** [renovatebot/renovate#24555] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.93.2

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

renovate able to have executions longer than 60 mins

### What you tried so far.

setup an NPM access token using https://www.npmjs.com/package/google-artifactregistry-auth, unfortunately they only last for 60 mins.  When renovate bot executions take less than the lifespan of the NPM access token, it runs fine and this is not an issue

### Relevant debug logs

<details><summary>Logs</summary>

```
https://us-npm.pkg.dev/my-project/my-npm-repository/path-parse (GET,403)
https://us-npm.pkg.dev/my-project/my-npm-repository/opossum (GET,403)
https://us-npm.pkg.dev/my-project/my-npm-repository/node-cache (GET,403)
https://us-npm.pkg.dev/my-project/my-npm-repository/moment (GET,403)
DEBUG: GET https://us-npm.pkg.dev/my-project/my-npm-repository/path-parse = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=403 retryCount=0, duration=26)
DEBUG: GET https://us-npm.pkg.dev/my-project/my-npm-repository/lodash = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=403 retryCount=0, duration=42)
DEBUG: GET https://us-npm.pkg.dev/my-project/my-npm-repository/mongodb-memory-server = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=403 retryCount=0, duration=61
```

</details>

**Comment by rvillane** [comment]
An option is have renovate implement [OAuth 2.0 to access Google APIs](https://developers.google.com/identity/protocols/oauth2#5.-refresh-the-access-token,-if-necessary.), however, I'm not clear if that option is supported by renovate at this moment.

Any ideas about how to configure long running executions against a private NPM registry that enforces authentication ?

**Comment by rarkins** [comment]
Do you mean?
- You generate a temporary token which lasts 60 minutes
- Renovate takes longer than 60 minutes to complete all repos

**Reply by rvillane** [reply]
yup, exactly, is a large list of repos when it happens

**Reply by rarkins** [reply]
I think this would require _Renovate_ to be capable of generating and then refreshing (if necessary) temporary tokens. This would mean that the original credentials need to be passed to Renovate so that it has the capability to do so.

**Reply by rvillane** [reply]
is such functionality available in renovate at this moment ?

**Reply by viceice** [reply, marked as answer]
no
